src/core/gui/PageView.cpp

```diff
@@ -890,6 +914,94 @@ void XojPageView::drawLoadingPage(cairo_t* cr) {
     rerenderPage();
 }
 
+bool XojPageView::displayLinkPopover(std::shared_ptr<XojPdfPage> page, double pageX, double pageY) {
+    // Search for selected link
+    const auto links = page->getLinks();
+
+    for (auto&& [rect, action]: links) {
+        std::shared_ptr<const LinkDestination> dest = action->getDestination();
+
+        if (!(rect.x1 <= pageX && pageX <= rect.x2 && rect.y1 <= pageY && pageY <= rect.y2)) {
+            continue;
+        }
+
+        GtkWidget* box = gtk_box_new(GTK_ORIENTATION_VERTICAL, 5);
+        GtkWidget* popover = makePopover(rect, box);
+
+        if (dest->isURI()) {
+            std::string uri = dest->getURI();
+            char* uriLabel = g_markup_escape_text(uri.c_str(), -1);
+
+            std::stringstream labelMarkup;
```

Comment:
`#include <sstream>`
is missing, I think. That's why it doesn't compile on MacOS.